packages/button/src/Button.tsx

```diff
@@ -360,7 +359,12 @@ const Button: ExtendableBox<BaseButtonProps, 'button'> = React.forwardRef(
     const modifiedGlyph =
       glyph && children
         ? React.cloneElement(glyph, {
-            className: cx({ [glyphMargins[size]]: glyph != null }),
+            className: cx(
+              css`
+                flex-shrink: 0;
```

Comment:
Ensures that the Icon doesn't shrink when the text overpowers it. We should consider building this into our icons perhaps, I feel like we apply this rule a lot.

Reply:
👍